File: Modelo/Sujeto.php
<?php
    namespace Modelo;
    require_once 'Observer.class.php';

    abstract class Sujeto
    {
        /**
         * 
         * @var Observer|array
         */
        protected $observadores = array();
        
        /**
         *
         * @param Observer $observador            
         */
        public function agrega(Observer $observador)
        {
            $this->observadores[] = $observador;
        }

        /**
         *
         * @param Observer $observador            
         */
        public function suprime(Observer $observador)
        {
            $key = array_search($observador, $this->observadores);
            if ($key !== FALSE)
            {
                unset($this->observadores[$key]);
            }
        }

        public function notifica()
        {
            foreach ($this->observadores as $observador)
            {
                $observador->actualiza();
            }
        }
    }
?>

File: appMedicoj/web/Modelo/Sujeto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Modelo;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Erick
 */
abstract class Sujeto {
    protected List<Observer> observadores = 
        new ArrayList<Observer>(); 
    
        public void agrega(Observer observador) 
    { 
        observadores.add(observador); 
    } 
 
    public void suprime(Observer observador) 
    { 
        observadores.remove(observador); 
    } 
}
